### PR TITLE
Remove preorder feature

### PR DESCRIPTION
**Pull request by Droniu**
I want to merge this change because it drops the preorder feature.

**PR intended to be tested with API branch:**  main (core can drop this separately)

## Impact
- Product create view
- Product update view
- Product variant create view
- Product variant update view
- Order return view (minor change, products in preorder couldn't be replaced so the checkbox wasn't rendered)
- Order fulfill view
- Order list view (filters)

## Pull Request Checklist

1. [x] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
MARKETPLACE_URL=https://marketplace-gray.vercel.app/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-8006 Remove preorder feature](https://app.clickup.com/t/2549495/SALEOR-8006)

**Review by karola312 — DISMISSED** [review]

**Review by dominik-zeglen — DISMISSED** [review]

File: src/orders/components/OrderFulfillLine/OrderFulfillLine.tsx [hunk 1]
```diff
@@ -86,14 +83,10 @@ export const OrderFulfillLine: React.FC<OrderFulfillLineProps> = props => {
         className={classes.colName}
         thumbnail={line?.thumbnail?.url}
         badge={
-          isPreorder || !line?.variant ? (
+          !line?.variant ? (
```
Comment (andrzejewsky) [inline src/orders/components/OrderFulfillLine/OrderFulfillLine.tsx:86]: This component (OrderFulfillLine.tsx) seems like a big one - if you touch this - how about splitting some parts into smaller units? Think of extracting components as well as functions that do something eg. display some text, prepare some event handlers and so on

File: src/orders/components/OrderFulfillLine/OrderFulfillLine.tsx [hunk 2]
```diff
@@ -110,54 +103,50 @@ export const OrderFulfillLine: React.FC<OrderFulfillLineProps> = props => {
...
+              [classes.quantityInnerInputNoRemaining]: !line.variant
+                ?.trackInventory,
+            }),
+            min: 0,
+            style: { textAlign: "right" },
+          }}
+          fullWidth
+          value={lineFormQuantity}
+          onChange={event =>
+            formsetChange(line.id, [
```
Comment (andrzejewsky) [inline src/orders/components/OrderFulfillLine/OrderFulfillLine.tsx:123]: eg. perhaps this TableCell deserves for own component? You could also prepare event handlers in a better way there.

File: src/orders/components/OrderFulfillLine/OrderFulfillLine.tsx [hunk 3]
```diff
@@ -110,54 +103,50 @@ export const OrderFulfillLine: React.FC<OrderFulfillLineProps> = props => {
...
+            ])
+          }
+          error={overfulfill}
+          variant="outlined"
+          InputProps={{
+            classes: {
+              ...(isStockExceeded &&
+                !overfulfill && {
+                  notchedOutline: classes.warning,
+                }),
```
Comment (andrzejewsky) [inline src/orders/components/OrderFulfillLine/OrderFulfillLine.tsx:137]: same here - is it readable? would be nice to "name" it by extracting to some field / function

File: src/orders/components/OrderFulfillPage/OrderFulfillPage.tsx [hunk 4]
```diff
@@ -180,15 +178,12 @@ const OrderFulfillPage: React.FC<OrderFulfillPageProps> = props => {
...
-      .filter(item => !!item?.value) // this can be removed after preorder is dropped
-      .some(item => {
-        const formQuantityFulfilled = item?.value?.[0]?.quantity;
-        const quantityToFulfill = order?.lines?.find(
-          line => line.id === item.id,
-        ).quantityToFulfill;
-        return formQuantityFulfilled > quantityToFulfill;
-      });
+    const overfulfill = formsetData.some(item => {
+      const formQuantityFulfilled = item?.value?.[0]?.quantity;
```
Comment (andrzejewsky) [inline src/orders/components/OrderFulfillPage/OrderFulfillPage.tsx:182]: a bit unsafe operations here, try to cover the scenario when there this statement is undefined

**Review by andrzejewsky — APPROVED** [review]
I'm giving approval here. I know the feature working, however if you have time, try to think about code-quality improvements: cover undefined scenarios, try to breakdown bigger components into smaller ones (one component - one responsibility), same with functions - when you use long statements perhaps worth to wrap/delegate it into function. If you do even one thing, it will be very welcome.

**Pull request closed without merging**